Sources/SwiftMetricsBluemix/SwiftMetricsBluemix.swift

```diff
@@ -99,76 +101,58 @@ public class AutoScalar {
   }
 
   private func initCredentials() -> Bool {
-    do {
-      let appEnv = try CloudFoundryEnv.getAppEnv()
-      var autoScalingServiceCreds: [String:Any] = [:]
-      // Find auto-scaling service using convenience method
-      let appEnvServices = appEnv.getServices()
-      for (_, service) in appEnvServices {
-        if service.label.hasPrefix(autoScalingServiceLabelPrefix) {
-          Log.debug("[Auto-Scaling Agent] Found Auto-Scaling service: \(service.name)")
-          autoScalingServiceCreds = service.credentials ?? [:]
-          break
-        }
-      }
-      Log.debug("[Auto-Scaling Agent] Auto-scaling service credentials: \(autoScalingServiceCreds)")
-
-      if autoScalingServiceCreds.isEmpty {
-        Log.error("[Auto-Scaling Agent] Please bind auto-scaling service!")
-        return false
-      }
-
-      // Validate optionals
-      guard let agentUsername = autoScalingServiceCreds["agentUsername"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.agentUsername is not found or empty.")
-        return false
-      }
-
-      guard let agentPassword = autoScalingServiceCreds["agentPassword"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.agentPassword is not found or empty.")
-        return false
-      }
-
-      guard let appID = autoScalingServiceCreds["app_id"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.app_id is not found or empty.")
-        return false
-      }
-
-      guard let host = autoScalingServiceCreds["url"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.url is not found or empty.")
-        return false
-      }
-
-      guard let serviceID = autoScalingServiceCreds["service_id"] as? String else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.service_id is not found or empty.")
-        return false
-      }
-
-      // Assign unwrapped values
-      self.host = host
-      self.serviceID = serviceID
-      self.appID = appID
-      self.agentPassword = agentPassword
-      self.agentUsername = agentUsername
-
-      guard let app = appEnv.getApp() else {
-        Log.warning("[Auto-Scaling Agent] sendMetrics:serviceEnv.url is not found or empty.")
-        return false
-      }
-
-      // Extract fields from App object
-      appName = app.name
-      instanceIndex = app.instanceIndex
-      instanceId = app.instanceId
+    let configMgr = ConfigurationManager().load(.environmentVariables)
+    // Find auto-scaling service using convenience method
+    let scalingServ: Service? = configMgr.getServices(type: autoScalingServiceLabel).first
```

Comment:
So the issue we have with getServices(type) is that it is possible to define an Auto-Scaling service without having to specify the type as "Auto-Scaling", according to @qibobo. See https://github.com/IBM-Swift/SwiftMetrics/issues/33 to see what he created - in that case I guess the type would be 'swift'. 
Also, from my reading of the code, isn't the parameter for AutoScalingService(withService) expecting a String rather than a Service?